.clang-format

```diff
@@ -5,14 +5,27 @@ AlignConsecutiveMacros: true
 AlignConsecutiveAssignments: true
 AlignConsecutiveDeclarations: true
 AlwaysBreakAfterReturnType: AllDefinitions
+# Can enable the following section when llvm 12.x is out
```

Comment:
We should keep track of the versions somehow (There were some version 11 changes that would be nice).
It will likely depend on the version of clang on the ubuntu runner provided github.

Reply:
We should probably update the format_source script to detect the version of clang-format that's installed and use the correct formatting file.   And maybe set a minimum version to use, so that our formatting is stable.